specification/attestation/data-plane/Microsoft.Attestation/stable/2020-10-01/attestation.json

```diff
@@ -0,0 +1,824 @@
+              "type": "string"
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          },
+          "401": {
+            "description": "Request is unauthorized",
+            "schema": {
+              "type": "string"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Add PolicyCertificate": {
+            "$ref": "./examples/Add_PolicyCertificate.json"
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "Policy Management Certificates"
+        ],
+        "summary": "Removes the specified policy management certificate. Note that the final policy management certificate cannot be removed.",
+        "operationId": "PolicyCertificates_Remove",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "policyCertificateToRemove",
+            "in": "body",
+            "description": "An RFC7519 JSON Web Token containing a claim named \"maa-policyCertificate\" whose value is an RFC7517 JSON Web Key which specifies a new key to update. The RFC7519 JWT must be signed with one of the existing signing certificates",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "An RFC7519 JSON Web Token structure containing an RFC7517 JsonWebKeySet which contains the certificates used to validate policy changes",
+            "schema": {
+              "type": "string"
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          },
+          "401": {
+            "description": "Request is unauthorized",
+            "schema": {
+              "type": "string"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Remove PolicyCertificate": {
+            "$ref": "./examples/Remove_PolicyCertificate.json"
+          }
+        }
+      }
+    },
+    "/attest/OpenEnclave": {
+      "post": {
+        "tags": ["attestation"],
+        "summary": "Attest to an SGX enclave.",
+        "description": "Processes an OpenEnclave report , producing an artifact. The type of artifact produced is dependent upon attestation policy.",
+        "operationId": "AttestOpenEnclave",
```

Comment:
Typically operation IDs are of the form `NOUN_VERB`.  The NOUN is used to generate a class/type name, and VERB is used to generate a method name.  If you omit the `NOUN_` portion then the code generators will synthesize one for you (it's usually something generic like `Operation`, and it might not be the same across all languages).  You can leave it as-is, I'm just pointing this out in case you want complete control over the naming.

Reply:
I've gone back and forth on this particular one. The core problem is that it gets really ugly from an API shape - there is only ever one verb, so we end up with 6 nouns, each of which has only a single verb attached. So you'd have SgxEnclave.Attest, OpenEnclave.Attest, SevSnpVm.Attest, VBSEnclave.Attest (note: This one is being renamed), etc. I'm open to making the change, it just seems awkward to me. The track1 code generator doesn't synthesize anything, not 100% sure what the track2 generators do though.

Reply:
I see your point.

For track 2, a lot (if not all) code generators will generate a generic noun if it's missing which is different from track 1 (coincidentally, we were discussing this in codegen scrum last week).  I think a lot of languages use `Operations` (we do for Go) but I'm not sure if this is 100% uniform.

Reply:
I think I'll go with Attestation as the noun and AttestXxxx as the method, I think that is more consistent with the preferred model.